histomicstk/cli/SuperpixelSegmentation/SuperpixelSegmentation.py

```diff
@@ -0,0 +1,225 @@
+        # images rather than inserting them.
+        vimg = pyvips.Image.new_from_memory(
+            numpy.ascontiguousarray(data).data,
+            data.shape[1], data.shape[0], data.shape[2],
+            large_image.constants.dtypeToGValue[data.dtype.char])
+        vimg = vimg.copy(interpretation=pyvips.Interpretation.RGB)
+        vimgTemp = pyvips.Image.new_temp_file('%s.v')
+        vimg.write(vimgTemp)
+        vimg = vimgTemp
+        x = tile['x'] - x0
+        ty = tile['tile_position']['region_y']
+        while len(strips) <= ty:
+            strips.append(None)
+        if strips[ty] is None:
+            strip = pyvips.Image.black(
+                tiparams.get('region', {}).get('width', meta['sizeX']),
+                vimg.height, bands=vimg.bands)
+            strip = strip.copy(interpretation=pyvips.Interpretation.RGB)
+            strips[ty] = [tile['y'] - y0, strip]
+        strips[ty][1] = strips[ty][1].composite([vimg], pyvips.BlendMode.OVER, x=int(x), y=0)
+    print('>> Found %d superpixels' % found)
+    if found > 256 ** 3:
+        print('Too many superpixels')
+    img = pyvips.Image.black(
+        tiparams.get('region', {}).get('width', meta['sizeX']),
+        tiparams.get('region', {}).get('height', meta['sizeY']),
+        bands=strips[0][1].bands)
+    img = img.copy(interpretation=pyvips.Interpretation.RGB)
+    for stripidx in range(len(strips)):
+        img = img.composite(
+            [strips[stripidx][1]], pyvips.BlendMode.OVER, x=0, y=int(strips[stripidx][0]))
+    # Discard alpha band, if any.
+    img = img[:3]
+    # Add program run parameters to the image description and list the
+    # superpixel count
+    img.set_type(
+        pyvips.GValue.gstr_type, 'image-description',
+        json.dumps(dict(vars(opts), indexCount=found)))
+    img.write_to_file(
+        opts.outputImageFile, tile=True, tile_width=256, tile_height=256, pyramid=True,
+        region_shrink='nearest',
+        bigtiff=True, compression='lzw', predictor='horizontal')
+
+    if opts.outputAnnotationFile:
+        print('>> Generating annotation file')
+        categories = [
+            {
+                'label': 'no_category',
+                'fillColor': 'rgba(0, 0, 0, 0)',
+                'strokeColor': 'rgba(0, 0, 0, 1)',
+            },
+            {
+                'label': opts.label_1,
+                'fillColor': opts.fillColor_1,
+                'strokeColor': opts.strokeColor_1,
+            },
+            {
+                'label': opts.label_2,
+                'fillColor': opts.fillColor_2,
+                'strokeColor': opts.strokeColor_2,
+            },
+            {
+                'label': opts.label_3,
+                'fillColor': opts.fillColor_3,
+                'strokeColor': opts.strokeColor_3,
+            },
+        ]
+        region_dict = utils.get_region_dict(opts.roi, None, ts)
+        annotation = {
+            'name': 'Superpixel Pixelmap %s' % (
```

Comment:
This ends up with a long, redundant default name.  We can drop the leading 'Superpixel Pixelmap', as that ends up in the default output name.  I think adding some parameters might be useful (e.g., `'%s - compactness %g sigma %g' % (os.path.splitext(os.path.basename(opts.outputAnnotationFile))[0]), compactness, sigma)`).